drivers/spi/spi_sam.c

```diff
@@ -155,7 +159,11 @@ static int spi_sam_configure(const struct device *dev,
 
 	regs->SPI_CR = SPI_CR_SPIDIS; /* Disable SPI */
 	regs->SPI_MR = spi_mr;
-	regs->SPI_CSR[config->slave] = spi_csr;
+	if (config->cs) {
+		regs->SPI_CSR[0] = spi_csr;
+	} else {
+		regs->SPI_CSR[config->slave] = spi_csr;
+	}
```

Comment:
NAK!

This not seems to be a correct, at first. `config->cs` is a list with no meaning here. For instance, let's assume `config->cs` define 2 GPIO lines but user will use on CS0 and CS3. How driver should attribute that?

This change seems to masked the real problem which I still not understood.

Reply:
As of #56576 `cs` is no longer a pointer...

Reply:
@pdgendt indeed, but not merged yet

Reply:
rebased and fixed

Reply:
Hi @teburd ,

Thank you for the details. Now it is more clear the whole issue.

Since this seems to be the correct solution I'll suggest to make the check before first use and store on a local variable. Then simple replace the **config->slave** on the two places instead add the condition.

Something like:


```diff
+	int cs_periph = spi_cs_is_gpio(config) ? 0 : config->slave;

-	spi_mr |= SPI_MR_PCS(spi_slave_to_mr_pcs(config->slave));
+	spi_mr |= SPI_MR_PCS(spi_slave_to_mr_pcs(cs_periph));
...
-	regs->SPI_CSR[config->slave] = spi_csr;
+	regs->SPI_CSR[cs_periph] = spi_csr;
```

Reply:
Done